DataFormats/EcalDigi/interface/EcalEBPhase2TriggerPrimitiveSample.h

```diff
@@ -0,0 +1,49 @@
+#ifndef ECALEBPHASE2TRIGGERPRIMITIVESAMPLE_H
+#define ECALEBPHASE2TRIGGERPRIMITIVESAMPLE_H 1
+
+#include <ostream>
+#include <cstdint>
+
+/** \class EcalEBPhase2TriggerPrimitiveSample
+\author N. Marinelli - Univ of Notre Dame
+
+*/
+
+class EcalEBPhase2TriggerPrimitiveSample {
+public:
+  EcalEBPhase2TriggerPrimitiveSample();
+  EcalEBPhase2TriggerPrimitiveSample(uint32_t data);
+  EcalEBPhase2TriggerPrimitiveSample(int encodedEt);
+  EcalEBPhase2TriggerPrimitiveSample(int encodedEt, bool isASpike);
+  EcalEBPhase2TriggerPrimitiveSample(int encodedEt, bool isASpike, int timing);
+
+  ///Set data
+  void setValue(uint32_t data) { theSample = data; }
+  // The sample is a 18 bit word defined as:
+  //
+  //     o o o o o    o     o o o o o o o o o o o o
+  //     |________|        |_______________________|
+  //      ~60ps res  spike         Et
+  //      time info  flag
+  //
+
+  /// get the raw word
+  uint32_t raw() const { return theSample & 0x3ffff; }
```

Comment:
If you do not mind, I will do it at a next iteration.  No-one objected to the same kind of writing in Phase I

Reply:
Thanks. It can come in a future update if it is not possible at this stage.
I do not know who reviewed the Phase 1 code but these kind of hard coded constants have made other code migrations to Phase 2 quite a tedious jobs and I would like to improve the situation for whatever future change that may be necessary. It also makes the code more readable.